src/main/java/org/opentripplanner/middleware/auth/Auth0Users.java

```diff
@@ -136,6 +139,31 @@ public static User getUserByEmail(String email, boolean createIfNotExists) {
         return null;
     }
 
+    /**
+     * Method to update a user's Auth0 metadata to include their preferred language as set in OTP-RR. This will
+     * ensure that account emails (such as email verification and reset password emails) are sent in the correct
+     * language.
+     */
+    public static void updateAuth0Language(OtpUser user, OtpUser preExistingUser) {
+        // Only call API if the user language has changed
+        if (!Objects.equals(user.preferredLocale, preExistingUser.preferredLocale)) {
+            try {
+                Map<String, Object> userMetadata = new HashMap<>();
+                userMetadata.put("lang", user.preferredLocale);
+                // Passing the entire user here will return a 403, because Auth0 sees it as you overwriting
+                // read-only fields, so create a blank user and give it only the data we want to update.
+                User auth0user = new User();
+                auth0user.setUserMetadata(userMetadata);
```

Comment:
Replace `userMetaData` with `Map.of("lang", user.preferredLocale)`

Reply:
Auth0 uses slightly different codes for certain languages (e.g. French: `fr` => `fr-FR`, Chinese Simplified/Traditional: `zh-Hans/t` => `zh-CN/TW`), so we will need to map them here, otherwise the messages for some of the languages will revert to English.

Reply:
> Replace `userMetaData` with `Map.of("lang", user.preferredLocale)`
Thanks, fixed in [a954131](https://github.com/ibi-group/otp-middleware/pull/346/commits/a954131009d4f11ac17afe731a5cb512b83f7807)